src/fu-engine.c

```diff
@@ -2190,7 +2203,25 @@ fu_engine_check_requirements(FuEngine *self,
 			XbNode *req = g_ptr_array_index(reqs, i);
 			if (!fu_engine_check_requirement(self, release, req, flags, error))
 				return FALSE;
+			if (fu_engine_requirement_is_specific(req))
+				has_specific_requirement = TRUE;
+		}
+	}
+
+	/* if a device uses a generic ID (i.e. not matching the OEM) then check to make sure the
+	 * firmware is specific enough, e.g. by using a CHID or depth requirement */
+	if (device != NULL &&
+	    fu_device_has_internal_flag(device, FU_DEVICE_INTERNAL_FLAG_ENFORCE_REQUIRES) &&
+	    !has_specific_requirement) {
+		if ((flags & FWUPD_INSTALL_FLAG_FORCE) == 0) {
```

Comment:
do we actually want to have a `--force` here?  I'd think that's a way to defeat security as `--force` can be added by the client.

Reply:
Maybe only in `SUPPORTED_BUILD=0`?

Reply:
I think `SUPPORTED_BUILD` is a great idea, fixed.